### PR TITLE
Report sub hypercube errors as errors unless calc cond

### PR DESCRIPTION
**Pull request by DnlLrssn**
**Checklist**

- [ ] tests added
- [x] commits conform to the [Commit message format](https://github.com/qlik-oss/gopherciser/blob/master/.github/CONTRIBUTING.md#commit)
- [ ] documentation updated

**Squash commit message**

Report errors on dimension and measure level as errors, unless it's a calculation condition.

**Info**

Optional informative text (not to be included in commit message) relevant to reviewers.

### CODE REVIEW

File: session/objecthandling.go [hunk 1]
```diff
@@ -706,44 +706,60 @@ func checkHyperCubeErrors(id string, hypercube *enigmahandlers.HyperCube, logEnt
...
 	if hypercube.DimensionInfo != nil {
 		for i, dimInfo := range hypercube.DimensionInfo {
-			if err := checkHyperCubeErrorInner(id, fmt.Sprintf("hypercube.DimensionInfo[%d]", i), dimInfo.Error); err != nil {
-				logEntry.Log(logger.WarningLevel, fmt.Sprintf("object<%s> has hypercube error<%s> in DimensionInfo[%d]", id, EngineCodeToString(dimInfo.Error.ErrorCode), i))
+			if err, warning := checkHyperCubeErrorInner(id, fmt.Sprintf("hypercube.DimensionInfo[%d]", i), dimInfo.Error); err != nil {
+				if !warning {
+					return errors.Wrapf(err, "object<%s> has hypercube error<%s> in DimensionInfo[%d]", id, EngineCodeToString(dimInfo.Error.ErrorCode), i)
+				}
+				if firstWarnError != nil {
+					firstWarnError = err
```
Comment (atluq) [inline session/objecthandling.go:721]: Can't we simply return not return here and in the other similar if statements?

Comment (DnlLrssn) [reply session/objecthandling.go:721]: the other non warn errors would not be found if there is an "error" which should only be warning

Comment (atluq) [reply session/objecthandling.go:721]: Right I get it!

Comment (DnlLrssn) [reply session/objecthandling.go:721]: yepp, I'll fix

Comment (DnlLrssn) [reply session/objecthandling.go:721]: fixed

**Review by atluq — APPROVED** [review]
Looks good!

**Review by atluq — APPROVED** [review]

**Pull request merged**